src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol_ImplementationChecks.cs

```diff
@@ -985,11 +994,13 @@ private void CheckNewModifier(Symbol symbol, bool isNew, DiagnosticBag diagnosti
 
                             for (int i = 0; i < overridingParameters.Length; i++)
                             {
-                                var overridenParameterType = overriddenParameters[i].TypeWithAnnotations;
+                                var overriddenParameterType = overriddenParameters[i].TypeWithAnnotations;
                                 var overridingParameterType = overridingParameters[i].TypeWithAnnotations;
-                                if (!overridingParameterType.Equals(overridenParameterType,
-                                                                         TypeCompareKind.AllIgnoreOptions & ~(TypeCompareKind.IgnoreNullableModifiersForReferenceTypes)) &&
-                                    overridingParameterType.Equals(overridenParameterType, TypeCompareKind.AllIgnoreOptions))
```

Comment:
>overridingParameterType.Equals(overridenParameterType, TypeCompareKind.AllIgnoreOptions) [](start = 36, length = 88)

I think we should keep this condition. Its purpose is to suppress diagnostics when return type doesn't match at all, this is an error condition, but I think we can get here in that case.